pyramid/request.py

```diff
@@ -214,10 +213,8 @@ def response(self):
         right" attributes (e.g. by calling ``request.response.set_cookie()``)
         within a view that uses a renderer.  Mutations to this response object
         will be preserved in the response sent to the client."""
-        registry = self.registry
-        response_factory = registry.queryUtility(IResponseFactory,
-                                                 default=Response)
-        return response_factory()
+        response_factory = _get_response_factory(self.registry)
+        return response_factory(self)
```

Comment:
The impl of `_get_response_factory` is identical to this code and defaults to `Response` if the user has not registered their own factory. I see no incompatibility here.